autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/configuration/AutofillRuntimeConfigProvider.kt

```diff
@@ -73,12 +74,16 @@ class RealAutofillRuntimeConfigProvider @Inject constructor(
     private fun determineIfAutofillEnabled(): Boolean =
         autofillStore.autofillAvailable && autofillStore.autofillEnabled && deviceAuthenticator.hasValidDeviceAuthentication()
 
-    private suspend fun determineIfCredentialsAvailable(url: String?): Boolean {
+    private suspend fun determineIfCredentialsAvailable(url: String?): AvailableInputTypeCredentials {
         return if (url == null || !determineIfAutofillEnabled()) {
-            false
+            AvailableInputTypeCredentials(username = false, password = false)
         } else {
             val savedCredentials = autofillStore.getCredentials(url)
-            savedCredentials.isNotEmpty()
+
+            val usernameSearch = savedCredentials.find { it.username != null && it.username!!.isNotBlank() }
```

Comment:
NIT: can't this be simplified to `!it.username.isNullOrEmpty()` and `!it.password.isNullOrEmpty()`